File: app/src/main/java/com/cst/androidlab/ApplicationController.java
package com.cst.androidlab;

import android.app.Application;

import androidx.room.Room;

import com.cst.androidlab.data.ToDoDataBase;

public class ApplicationController extends Application {
    private static ApplicationController instance;
    private static ToDoDataBase toDoDataBase;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        setupDataBase();
    }

    private void setupDataBase() {
        String toDoDataBaseName = "ToDoDB";

        toDoDataBase = Room.databaseBuilder(
                getApplicationContext(),
                ToDoDataBase.class,
                        toDoDataBaseName)
//                .addMigrations(ToDoDataBase.MIGRATION_1_2)
                // .fallbackToDestructiveMigration()
                .build();
    }

    public static ToDoDataBase getToDoDataBase() {
        return toDoDataBase;
    }

    public static ApplicationController getInstance() {
        return instance;
    }
}

File: app/src/main/java/com/cst/cstacademy2024/ApplicationController.kt
package com.cst.cstacademy2024

import android.app.Application
import androidx.room.Room
import com.cst.cstacademy2024.data.AppDatabase

class ApplicationController : Application() {

    lateinit var appDatabase: AppDatabase
        private set

    companion object {
        lateinit var instance: ApplicationController
            private set
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
        setUpDatabase()
    }


    private fun setUpDatabase() {
        appDatabase = Room.databaseBuilder(
            applicationContext,
            AppDatabase::class.java,
            "cst-academy-db"
        ).build()

    }
}